code/game/objects/items/weapons/twohanded.dm

```diff
@@ -432,3 +432,68 @@
 			M.faction = user.faction.Copy()
 			M.Copy_Parent(user, 100, user.health/2.5, 12, 30)
 			M.GiveTarget(L)
+			
+/obj/item/weapon/twohanded/ram
+	name = "kinetic battering ram"
+	desc = "Internal gyroscopes give this ram excellent balance and the necessary force to break doors... or ribs."
+	icon_state = "ram0"
+	force = 10
+	force_wielded = 24
+	w_class = 4
+	throwforce = 15
+	throw_speed = 2
+	throw_range = 3
+	slot_flags = SLOT_BACK
+	materials = list(MAT_METAL=13000)
+	origin_tech = "materials=2;engineering=2;combat=2"
+	attack_verb = list("crushed", "bashed", "bludgeoned", "smashed", "pulverized")
+	hitsound = "swing_hit"
+	var/removingairlock = 0
+
+/obj/item/weapon/twohanded/ram/update_icon()  
+	icon_state = "ram[wielded]"
+	return
+
+/obj/item/weapon/twohanded/ram/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] swings [src] into their face! It looks like they're trying to commit suicide..</span>")
+	playsound(src,'sound/effects/bang.ogg',50,1)
+	return (BRUTELOSS)
+
+/obj/item/weapon/twohanded/ram/afterattack(atom/T, mob/user, proximity)
+	if(!proximity) 
+		return
+	if(!wielded)
+		src << "<span class='warning'>You need both hands to break down a door!</span>"
```

Comment:
This message is gonna display anytime you click on anything with the battering ram